pkg/networking/middleware/capture.go

```diff
@@ -0,0 +1,16 @@
+package middleware
+
+import "net/http"
+
+// HeaderCapture A middleware that captures the headers of the request and doesn't send it
+type HeaderCapture struct {
```

Comment:
Probably used for testing?

Reply:
I use it in "AddHeaders". 

Previously we were calling functions defined in the middlewares that add headers to a request one by one. This is more generic, I run the request through all the MW and collect all the headers that were added without needing both a "RoundTrip" and an "AddXHeaders" function for every MW.

But yeah it's also useful for testing :)